### PR TITLE
Dockerfile for RPM packaging, based on CentOS 7

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
Fix for issue https://github.com/dotnet/dotnet-buildtools-prereqs-docker/issues/92

### CODE REVIEW

File: src/centos/7/rpmpkg/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7
+
+# Install the rpm packaging toolchain we need to build rpms
+RUN yum install -y \
+        gcc \
+        make \
+        rpm-build \
+        ruby-devel \
+        rubygems \
+        sudo \
```
Comment (MichaelSimons) [inline src/centos/7/rpmpkg/Dockerfile]: Should this be doing a `yum clean all` at the end?

File: src/centos/7/rpmpkg/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7
+
+# Install the rpm packaging toolchain we need to build rpms
+RUN yum install -y \
+        gcc \
+        make \
```
Comment (MichaelSimons) [inline src/centos/7/rpmpkg/Dockerfile]: `make` and `sudo` are already installed in the base image.

File: src/centos/7/rpmpkg/Dockerfile (hunk 1 above)
Comment (NikolaMilosavljevic) [reply src/centos/7/rpmpkg/Dockerfile]: I'm not sure - I've used this file as model: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/master/src/centos/7/mlnet/Dockerfile

@dagood you had specified "yum clean" commands in the suggested solution - issue https://github.com/dotnet/dotnet-buildtools-prereqs-docker/issues/92. Are those commands needed?

File: src/centos/7/rpmpkg/Dockerfile (hunk 2 above)
Comment (NikolaMilosavljevic) [reply src/centos/7/rpmpkg/Dockerfile]: I'll remove them. Didn't seem to make a difference, but want to follow the best practices.

File: src/centos/7/rpmpkg/Dockerfile (hunk 1 above)
Comment (dagood) [reply src/centos/7/rpmpkg/Dockerfile]: It's a guideline, but yes, the `yum` cache data is not useful for the image user and should be cleaned up in the same `RUN` statement as the install commands. The user either won't run any more `yum` commands and have no use of a cache, but even if they do install more packages, they will probably find the cache data is old and useless.

File: src/centos/7/rpmpkg/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM mcr.microsoft.com/dotnet-buildtools/prereqs:centos-7
+
+# Install the rpm packaging toolchain we need to build rpms
```
Comment (MichaelSimons) [inline src/centos/7/rpmpkg/Dockerfile]: Nit - consider avoiding `we` in the comment - maybe something like `Install the rpm packaging toolchain required to build rpms

Comment (NikolaMilosavljevic) [reply src/centos/7/rpmpkg/Dockerfile]: Fixed.

**Pull request merged**